crates/rattler_conda_types/src/repo_data/mod.rs

```diff
@@ -315,6 +323,78 @@ impl PackageRecord {
     pub fn sort_topologically<T: AsRef<PackageRecord> + Clone>(records: Vec<T>) -> Vec<T> {
         topological_sort::sort_topologically(records)
     }
+
+    /// Validate that the given package records are valid w.r.t. 'depends' and 'constrains'.
+    /// This function will return Ok(()) if all records form a valid environment, i.e., all dependencies
+    /// of each package are satisfied by the other packages in the list.
+    /// If there is a dependency that is not satisfied, this function will return an error.
+    pub fn validate_package_records<T: AsRef<PackageRecord>>(
```

Comment:
Since this is already implemented in the PackageRecord maybe remove package_record from the name? Validate consistent maybe

Reply:
[`9e9dd97` (#911)](https://github.com/conda/rattler/pull/911/commits/9e9dd970e1f720dc901b25e715b059b39b41dfba)